test/jdk/java/foreign/TestSegments.java

```diff
@@ -380,12 +379,27 @@ void testReinterpret() {
         try (Arena arena = Arena.ofConfined()){
             // check size
             assertEquals(MemorySegment.ofAddress(42).reinterpret(100).byteSize(), 100);
+            assertEquals(MemorySegment.ofAddress(42).reinterpret(JAVA_INT).byteSize(), JAVA_INT.byteSize());
             assertEquals(MemorySegment.ofAddress(42).reinterpret(100, Arena.ofAuto(), null).byteSize(), 100);
             // check scope and cleanup
             assertEquals(MemorySegment.ofAddress(42).reinterpret(100, arena, s -> counter.incrementAndGet()).scope(), arena.scope());
             assertEquals(MemorySegment.ofAddress(42).reinterpret(arena, s -> counter.incrementAndGet()).scope(), arena.scope());
+            assertEquals(MemorySegment.ofAddress(42).reinterpret(JAVA_INT, arena, _ -> counter.incrementAndGet()).scope(), arena.scope());
+        }
+        assertEquals(counter.get(), 3);
+    }
+
+    @Test
+    void testReinterpretAlignment() {
+        AtomicInteger counter = new AtomicInteger();
+        try (Arena arena = Arena.ofConfined()){
+            var segment = MemorySegment.ofAddress(41).reinterpret(JAVA_INT);
```

Comment:
i.e. this call (and the one below) should throw an exception.

Reply:
Note - this means that the javadoc of the new methods should also document an IllegalArgumentException.